Makefile.Linux: use cc -dumpmachine

stokito opened the pull request:
Replace `$(CC) --version | grep "Target:" | cut -d ' ' -f 2)` with a simple `cc -dumpmachine`.
Actually the line above `CC := clang` is not needed anymore because we can build with gcc (and probably other compilers like Intel's icc).

Also I added a small improvement to getting_started.md

stokito committed:
getting_started.md Improve options table readability

Make the table looking better even in raw text mode
Replace “” quotes with ""

aajain-com commented on Build/Makefile.Linux:
```diff
@@ -1,7 +1,7 @@
 CC := clang
 AR := ar
-COMPILER := $(shell $(CC) --version | grep "Target:" | cut -d ' ' -f 2)
+COMPILER := $(shell $(CC) -dumpmachine)
```
This works with clang so we can potentially update the Makefile.Darwin to use the same command.

stokito replied:
yeah, I just afraid to do this because can't test it. But what about to remove the `CC := clang` variable and use system's cc?

aajain-com replied:
`clang -dumpmachine` seems to be working for me. But let's leave clang as the default compiler for Darwin. Will do some more testing on my end and then merge the change. Thanks for this change.

aajain-com replied:
@stokito I got a chance to do more testing and this works as intended for both `clang` and `cc`. Could you please make the similar change for Makefile.Darwin and we can then merge this change?

stokito replied:
2 min

stokito replied:
ok, I made the same change for Darwin and Raspi and made a force push

stokito committed:
Makefile.Linux: use cc -dumpmachine

The -dumpmachine flag returns a platform and this is exactly the same as clang --version returns in "Target:" line.
Also the same flag works for other compilers:

$ gcc -dumpmachine
x86_64-linux-gnu
$ clang -dumpmachine
x86_64-pc-linux-gnu

aajain-com commented:
Not sure why CI didn't run on this build. Closing and re-opening the PR to see if this kicks off the CI builds.

stokito commented:
maybe because I made a force push

aajain-com commented:
> maybe because I made a force push

No, it seems like the CI is turned off completely for this repo. Working on getting it enabled. Will merge your change once the CI is back up and green.
